web/src/compose_actions.ts

```diff
@@ -329,8 +329,16 @@ export let start = (raw_opts: ComposeActionsStartOpts): void => {
         compose_state.set_compose_recipient_id(compose_state.DIRECT_MESSAGE_ID);
         compose_recipient.on_compose_select_recipient_update();
     } else if (opts.stream_id) {
-        compose_state.set_stream_id(opts.stream_id);
-        compose_recipient.on_compose_select_recipient_update();
+        const stream = stream_data.get_sub_by_id(opts.stream_id);
+        if (stream && stream_data.can_post_messages_in_stream(stream)) {
+            compose_state.set_stream_id(opts.stream_id);
+            compose_recipient.on_compose_select_recipient_update();
+        } else {
+            opts.stream_id = undefined;
```

Comment:
From the issue description, it looks like we only want to do this if user clicks on "start new conversation" button.

Reply:
Also can we do similar to what do when user press `c` hotkey instead of this hack.

Reply:
> Also can we do similar to what do when user press `c` hotkey instead of this hack.

No, I don't think we can do that since `c` hotkey does exactly what the `start new conversation` button does and I think we need to update that behaviour in this issue.

Reply:
What happens when you click on a message or press r key?

Reply:
> What happens when you click on a message or press the `r` key?

Hey @amanagr,  
I found that pressing `r` does the same thing as the `c` hotkey.

Reply:
> From the issue description, it looks like we only want to do this if user clicks on "start new conversation" button.

Reply:
@amanagr, as I understand this issue, we need to modify the behavior of the `Start New Conversation` button for users who do not have permission to send messages to a particular stream.

I found that the `c` and `r` hotkeys are used to perform the same action as the `Start New Conversation` button. Since I have updated the behavior of that button, the behavior of these hotkeys will also be affected.

I don't think we want the behavior of these hotkeys to differ from that of the `Start New Conversation` button. If I am misunderstanding something, please help me clarify this.

Thanks a lot!

Reply:
Thanks @amanagr for pointing out this minor bug which I missed.
Now, I have updated the code in such a way that the new behaviour only gets triggered when someone clicks one `Start new conversation` button or use the hotkey `c` for the same.

The hotkey `r` now opens up the compose box as per the previous behaviour for replying to a topic.

Please let me know it any further modifications are required.

Thanks!